src/mechanics/IGA/NURBS.h

```diff
@@ -0,0 +1,540 @@
+#pragma once
+
+#include "mechanics/nodes/NodeSimple.h"
+#include "base/Exception.h"
+#include <array>
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+#include <vector>
+
+namespace NuTo
+{
+//! @author Peter Otto, BAM
+//! @date September, 2017
+//! @brief ... class for NURBS curves, with IGA specific functions
+//! @brief ... NURBS specific algorithms taken from Piegl, Tiller 'The NURBS book' 1996
+//! @brief ... TDimParameter is the dimension of the parametric space: curve is 1D and surface is 2D
+template <int TDimParameter>
+class NURBS
+{
+public:
+    enum mParametrization
+    {
+        chord,
+        centripetal
+    };
+
+    /** Constructors **/
+
+    //! @brief ... constructor
+    //! @param rDegree ... degree of the polynomial
+    //! @param rKnots ... knot vector
+    //! @param rControlPoints ... control points
+    NURBS(const std::array<std::vector<double>, TDimParameter>& knots,
+          const std::vector<std::vector<NodeSimple>>& controlPoints, const std::vector<std::vector<double>>& weights,
```

Comment:
This class could provide a very general NURBS definition. So I would ask you to save the control points in some other format (Eigen::VectorXd or Eigen::Matrix<double, TDimParameter, 1>?) to decouple it from `NodeSimple`.

Reply:
If we do that, I think we should move it to `math/`.  Seems like the more appropriate location.